### ISSUE TITLE
qulacs-gpu not working on Windows 10

### ISSUE THREAD

**Issue by sg495** [qulacs/qulacs#282]
I tried installing`qulacs-gpu`, both from `pip` and compiling it by hand, and I always get the following error on import:

```
Python 3.9.2 (tags/v3.9.2:1a79785, Feb 19 2021, 13:44:55) [MSC v.1928 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import qulacs
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: DLL load failed while importing qulacs: The specified module could not be found.
```

No error is raised if I install `qulacs` instead.

The CUDA (v11.2.2) folder is in Path and all relevant DLLs are reachable (I ran `dumpbin` from a VisualStudio terminal and checked manually). I am on Windows 10 Home 20H2 with an NVIDIA GeForce GTX 1660 Ti.

**Comment by corryvrequan** [comment]
@sg495 
Thanks for reporting a problem.
If all the relevant DLLs are reachable, and qulacs without GPU works, basically it should work, I guess. It is mysterious for me...

A possible reason is that environmental variables at compilation and those at execution are different.
Can you try the following check?

1. Open "./script/build_msvc_2017_with_gpu.bat" with an editor. Replace "Visual Studio 15 2017 Win64" to "Visual Studio 16 2019".
2. Execute "./script/build_msvc_2017_with_gpu.bat" from "./qulacs/"
3. You would see compile messages. In the messages, please verify related python version is "3.9.2" if you installed many python versions.
4. Move to "./qulacs/build/" and execute "make test". Please verify tests for GPU passes.
5. Also execute "make pythontest" (if this test passes, your DLL can be at least imported in a certain way.)

If an error happens during the above procedure, I hope the error message will illustrate what is wrong.

**Comment by sg495** [comment]
Thanks for such a quick and detailed reply, @corryvrequan! Step 3 succeeded and python version is 3.9.2. Step 4 reports a number of errors, full output below.

```
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  csim_static.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\lib\csim_static.lib
  gpusim_static.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\lib\gpusim_static.lib
  cppsim_static.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\lib\cppsim_static.lib
  gtest.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\visualstudio\_deps\googletest_fetch-src\googlemock\gtest\Release\gtest.lib
  gtest_main.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\visualstudio\_deps\googletest_fetch-src\googlemock\gtest\Release\gtest_main.lib
  cppsim_test.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\bin\cppsim_test.exe
  csim_test.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\bin\csim_test.exe
  gpusim_test.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\bin\gpusim_test.exe
  vqcsim_static.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\lib\vqcsim_static.lib
  vqcsim_test.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\bin\vqcsim_test.exe
  Running main() from C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\_deps\googletest_fetch-src\googletest\src\gtest_main.cc
  [==========] Running 46 tests from 3 test cases.
  [----------] Global test environment set-up.
  [----------] 4 tests from MemoryOperationTest
  [ RUN      ] MemoryOperationTest.AllocateAndRelease
  [       OK ] MemoryOperationTest.AllocateAndRelease (0 ms)
  [ RUN      ] MemoryOperationTest.MemoryZeroCheck
  [       OK ] MemoryOperationTest.MemoryZeroCheck (0 ms)
  [ RUN      ] MemoryOperationTest.HaarRandomState
  [       OK ] MemoryOperationTest.HaarRandomState (1 ms)
  [ RUN      ] MemoryOperationTest.LargeMemory
  [       OK ] MemoryOperationTest.LargeMemory (3 ms)
  [----------] 4 tests from MemoryOperationTest (4 ms total)

  [----------] 13 tests from StatOperationTest
  [ RUN      ] StatOperationTest.ProbTest
  [       OK ] StatOperationTest.ProbTest (4 ms)
  [ RUN      ] StatOperationTest.MarginalProbTest
  [       OK ] StatOperationTest.MarginalProbTest (2 ms)
  [ RUN      ] StatOperationTest.EntropyTest
  [       OK ] StatOperationTest.EntropyTest (0 ms)
  [ RUN      ] StatOperationTest.InnerProductTest
  [       OK ] StatOperationTest.InnerProductTest (1 ms)
  [ RUN      ] StatOperationTest.SingleQubitExpectationValueTest
  [       OK ] StatOperationTest.SingleQubitExpectationValueTest (3 ms)
  [ RUN      ] StatOperationTest.MultiQubitExpectationValueWholeTest
  [       OK ] StatOperationTest.MultiQubitExpectationValueWholeTest (1 ms)
  [ RUN      ] StatOperationTest.MultiQubitExpectationValueZopWholeTest
  [       OK ] StatOperationTest.MultiQubitExpectationValueZopWholeTest (2 ms)
  [ RUN      ] StatOperationTest.MultiQubitExpectationValuePartialTest
  [       OK ] StatOperationTest.MultiQubitExpectationValuePartialTest (1 ms)
  [ RUN      ] StatOperationTest.MultiQubitExpectationValueZopPartialTest
  [       OK ] StatOperationTest.MultiQubitExpectationValueZopPartialTest (2 ms)
  [ RUN      ] StatOperationTest.MultiQubitTransitionAmplitudeWholeTest
  [       OK ] StatOperationTest.MultiQubitTransitionAmplitudeWholeTest (2 ms)
  [ RUN      ] StatOperationTest.MultiQubitTransitionAmplitudeZopWholeTest
  [       OK ] StatOperationTest.MultiQubitTransitionAmplitudeZopWholeTest (2 ms)
  [ RUN      ] StatOperationTest.MultiQubitTransitionAmplitudePartialTest
  [       OK ] StatOperationTest.MultiQubitTransitionAmplitudePartialTest (2 ms)
  [ RUN      ] StatOperationTest.MultiQubitTransitionAmplitudeZopPartialTest
  [       OK ] StatOperationTest.MultiQubitTransitionAmplitudeZopPartialTest (2 ms)
  [----------] 13 tests from StatOperationTest (24 ms total)

  [----------] 29 tests from UpdateTest
  [ RUN      ] UpdateTest.SingleQubitControlSingleQubitDenseMatrixTest
  [       OK ] UpdateTest.SingleQubitControlSingleQubitDenseMatrixTest (59 ms)
  [ RUN      ] UpdateTest.TwoQubitControlSingleQubitDenseMatrixTest
  [       OK ] UpdateTest.TwoQubitControlSingleQubitDenseMatrixTest (63 ms)
  [ RUN      ] UpdateTest.SingleQubitControlTwoQubitDenseMatrixTest
  [       OK ] UpdateTest.SingleQubitControlTwoQubitDenseMatrixTest (6 ms)
  [ RUN      ] UpdateTest.TwoQubitControlTwoQubitDenseMatrixTest
  [       OK ] UpdateTest.TwoQubitControlTwoQubitDenseMatrixTest (11 ms)
  [ RUN      ] UpdateTest.SingleDenseMatrixTest
  [       OK ] UpdateTest.SingleDenseMatrixTest (3 ms)
  [ RUN      ] UpdateTest.ThreeQubitDenseMatrixTest
  [       OK ] UpdateTest.ThreeQubitDenseMatrixTest (38 ms)
  [ RUN      ] UpdateTest.TwoQubitDenseMatrixTest
  [       OK ] UpdateTest.TwoQubitDenseMatrixTest (90 ms)
  [ RUN      ] UpdateTest.SingleDiagonalMatrixTest
  [       OK ] UpdateTest.SingleDiagonalMatrixTest (1 ms)
  [ RUN      ] UpdateTest.SinglePhaseGateTest
  [       OK ] UpdateTest.SinglePhaseGateTest (2 ms)
  [ RUN      ] UpdateTest.MultiQubitDiagonalMatrixTest
  [       OK ] UpdateTest.MultiQubitDiagonalMatrixTest (1 ms)
  [ RUN      ] UpdateTest.MultiQubitDiagonalMatrixTest2
  [       OK ] UpdateTest.MultiQubitDiagonalMatrixTest2 (0 ms)
  [ RUN      ] UpdateTest.TwoQubitControlTwoQubitDiagonalMatrixTest
  [       OK ] UpdateTest.TwoQubitControlTwoQubitDiagonalMatrixTest (14 ms)
  [ RUN      ] UpdateTest.XGate
  [       OK ] UpdateTest.XGate (3 ms)
  [ RUN      ] UpdateTest.YGate
  [       OK ] UpdateTest.YGate (3 ms)
  [ RUN      ] UpdateTest.ZGate
  [       OK ] UpdateTest.ZGate (2 ms)
  [ RUN      ] UpdateTest.HGate
  [       OK ] UpdateTest.HGate (2 ms)
  [ RUN      ] UpdateTest.SGate
  [       OK ] UpdateTest.SGate (0 ms)
  [ RUN      ] UpdateTest.TGate
  [       OK ] UpdateTest.TGate (0 ms)
  [ RUN      ] UpdateTest.sqrtXGate
  [       OK ] UpdateTest.sqrtXGate (0 ms)
  [ RUN      ] UpdateTest.sqrtYGate
  [       OK ] UpdateTest.sqrtYGate (0 ms)
  [ RUN      ] UpdateTest.ProjectionAndNormalizeTest
  [       OK ] UpdateTest.ProjectionAndNormalizeTest (17 ms)
  [ RUN      ] UpdateTest.SingleQubitRotationGateTest
CUSTOMBUILD : error : gate::CNOT(UINT, UINT): control_qubit_index and target_qubit_index has the same value. [C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\test.vcxproj]
CUSTOMBUILD : error : gate::CZ(UINT, UINT): control_qubit_index and target_qubit_index has the same value. [C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\test.vcxproj]
CUSTOMBUILD : error : gate::SWAP(UINT, UINT): two indices have the same value. [C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\test.vcxproj]
CUSTOMBUILD : error : gate::Pauli(std::vector<UINT> target, std::vector<UINT>pauli_id): target list contains duplicated values. [C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\test.vcxproj]
CUSTOMBUILD : error : gate::PauliRotation(std::vector<UINT> target, std::vector<UINT>pauli_id, double angle): target list contains duplicated values. [C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\test.
vcxproj]
CUSTOMBUILD : error : gate::DenseMatrix(std::vector<UINT> target_list, ComplexMatrix matrix): target list contains duplicated values. [C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\test.vcxproj]
CUSTOMBUILD : error : gate::SparseMatrix(std::vector<UINT> target_list, SparseComplexMatrix matrix): target list contains duplicated values. [C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\test.vcxproj]
CUSTOMBUILD : error : gate::RandomUnitary(std::vector<UINT> target_list): target list contains duplicated values. [C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\test.vcxproj]
CUSTOMBUILD : error : gate::ReversibleBoolean(std::vector<UINT> target_qubit_index_list, std::function<ITYPE(ITYPE,ITYPE)> function_ptr): target list contains duplicated values. [C:\Users\sg495\Documents\git\qula
cs\qulacs\visualstudio\test.vcxproj]
CUSTOMBUILD : error : gate::TwoQubitDepolarizingNoise(UINT, UINT, double): target list contains duplicated values. [C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\test.vcxproj]
  [       OK ] UpdateTest.SingleQubitRotationGateTest (1 ms)
  [ RUN      ] UpdateTest.CNOTGate
  [       OK ] UpdateTest.CNOTGate (0 ms)
  [ RUN      ] UpdateTest.CZGate
  [       OK ] UpdateTest.CZGate (1 ms)
  [ RUN      ] UpdateTest.SWAPGate
  [       OK ] UpdateTest.SWAPGate (0 ms)
  [ RUN      ] UpdateTest.SingleQubitPauliTest
  [       OK ] UpdateTest.SingleQubitPauliTest (1 ms)
  [ RUN      ] UpdateTest.SingleQubitPauliRotationTest
  [       OK ] UpdateTest.SingleQubitPauliRotationTest (0 ms)
  [ RUN      ] UpdateTest.MultiQubitPauliTest
  [       OK ] UpdateTest.MultiQubitPauliTest (1 ms)
  [ RUN      ] UpdateTest.MultiQubitPauliRotationTest
  [       OK ] UpdateTest.MultiQubitPauliRotationTest (1 ms)
  [----------] 29 tests from UpdateTest (320 ms total)

  [----------] Global test environment tear-down
  [==========] 46 tests from 3 test cases ran. (348 ms total)
  [  PASSED  ] 46 tests.
  Running main() from C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\_deps\googletest_fetch-src\googletest\src\gtest_main.cc
  [==========] Running 87 tests from 9 test cases.
  [----------] Global test environment set-up.
  [----------] 12 tests from CircuitTest
  [ RUN      ] CircuitTest.CircuitBasic
  [       OK ] CircuitTest.CircuitBasic (2 ms)
  [ RUN      ] CircuitTest.CircuitOptimize
  [       OK ] CircuitTest.CircuitOptimize (0 ms)
  [ RUN      ] CircuitTest.RandomCircuitOptimize
  [       OK ] CircuitTest.RandomCircuitOptimize (5 ms)
  [ RUN      ] CircuitTest.RandomCircuitOptimize2
  [       OK ] CircuitTest.RandomCircuitOptimize2 (15 ms)
  [ RUN      ] CircuitTest.RandomCircuitOptimize3
  [       OK ] CircuitTest.RandomCircuitOptimize3 (21 ms)
  [ RUN      ] CircuitTest.SuzukiTrotterExpansion
  [       OK ] CircuitTest.SuzukiTrotterExpansion (3 ms)
  [ RUN      ] CircuitTest.RotateDiagonalObservable
  [       OK ] CircuitTest.RotateDiagonalObservable (1 ms)
  [ RUN      ] CircuitTest.SpecialGatesToString
  [       OK ] CircuitTest.SpecialGatesToString (0 ms)
  [ RUN      ] CircuitTest.CircuitOptimizeLight
  [       OK ] CircuitTest.CircuitOptimizeLight (0 ms)
  [ RUN      ] CircuitTest.RandomCircuitOptimizeLight
  [       OK ] CircuitTest.RandomCircuitOptimizeLight (1 ms)
  [ RUN      ] CircuitTest.RandomCircuitOptimizeLight2
  [       OK ] CircuitTest.RandomCircuitOptimizeLight2 (0 ms)
  [ RUN      ] CircuitTest.RandomCircuitOptimizeLight3
  [       OK ] CircuitTest.RandomCircuitOptimizeLight3 (1 ms)
  [----------] 12 tests from CircuitTest (49 ms total)

  [----------] 24 tests from GateTest
  [ RUN      ] GateTest.ApplySingleQubitGate
  [       OK ] GateTest.ApplySingleQubitGate (3 ms)
  [ RUN      ] GateTest.ApplySingleQubitRotationGate
  [       OK ] GateTest.ApplySingleQubitRotationGate (0 ms)
  [ RUN      ] GateTest.ApplyTwoQubitGate
  [       OK ] GateTest.ApplyTwoQubitGate (0 ms)
  [ RUN      ] GateTest.ApplyMultiQubitGate
  [       OK ] GateTest.ApplyMultiQubitGate (0 ms)
  [ RUN      ] GateTest.MergeTensorProduct
  [       OK ] GateTest.MergeTensorProduct (0 ms)
  [ RUN      ] GateTest.MergeMultiply
  [       OK ] GateTest.MergeMultiply (0 ms)
  [ RUN      ] GateTest.MergeTensorProductAndMultiply
  [       OK ] GateTest.MergeTensorProductAndMultiply (0 ms)
  [ RUN      ] GateTest.RandomPauliMerge
  [       OK ] GateTest.RandomPauliMerge (2 ms)
  [ RUN      ] GateTest.RandomPauliRotationMerge
  [       OK ] GateTest.RandomPauliRotationMerge (1 ms)
  [ RUN      ] GateTest.RandomUnitaryMerge
  [       OK ] GateTest.RandomUnitaryMerge (1 ms)
  [ RUN      ] GateTest.RandomUnitaryMergeLarge
  [       OK ] GateTest.RandomUnitaryMergeLarge (1 ms)
  [ RUN      ] GateTest.U3MergeIBMQGate
  [       OK ] GateTest.U3MergeIBMQGate (0 ms)
  [ RUN      ] GateTest.ControlMerge
  [       OK ] GateTest.ControlMerge (0 ms)
  [ RUN      ] GateTest.RandomControlMergeSmall
  [       OK ] GateTest.RandomControlMergeSmall (0 ms)
  [ RUN      ] GateTest.RandomControlMergeLarge
  [       OK ] GateTest.RandomControlMergeLarge (1 ms)
  [ RUN      ] GateTest.ProbabilisticGate
  [       OK ] GateTest.ProbabilisticGate (0 ms)
  [ RUN      ] GateTest.CPTPGate
  [       OK ] GateTest.CPTPGate (4 ms)
  [ RUN      ] GateTest.InstrumentGate
  [       OK ] GateTest.InstrumentGate (3 ms)
  [ RUN      ] GateTest.AdaptiveGate
  [       OK ] GateTest.AdaptiveGate (0 ms)
  [ RUN      ] GateTest.GateAdd
  [       OK ] GateTest.GateAdd (0 ms)
  [ RUN      ] GateTest.RandomUnitaryGate
  [       OK ] GateTest.RandomUnitaryGate (2 ms)
  [ RUN      ] GateTest.ReversibleBooleanGate
  [       OK ] GateTest.ReversibleBooleanGate (0 ms)
  [ RUN      ] GateTest.TestNoise
  [       OK ] GateTest.TestNoise (0 ms)
  [ RUN      ] GateTest.DuplicateIndex
  [       OK ] GateTest.DuplicateIndex (1 ms)
  [----------] 24 tests from GateTest (19 ms total)

  [----------] 17 tests from DensityMatrixGateTest
  [ RUN      ] DensityMatrixGateTest.ApplySingleQubitGate
  [       OK ] DensityMatrixGateTest.ApplySingleQubitGate (10 ms)
  [ RUN      ] DensityMatrixGateTest.ApplySingleQubitRotationGate
  [       OK ] DensityMatrixGateTest.ApplySingleQubitRotationGate (2 ms)
  [ RUN      ] DensityMatrixGateTest.ApplyTwoQubitGate
  [       OK ] DensityMatrixGateTest.ApplyTwoQubitGate (1 ms)
  [ RUN      ] DensityMatrixGateTest.ApplyMultiQubitGate
  [       OK ] DensityMatrixGateTest.ApplyMultiQubitGate (11 ms)
  [ RUN      ] DensityMatrixGateTest.MergeTensorProduct
  [       OK ] DensityMatrixGateTest.MergeTensorProduct (1 ms)
  [ RUN      ] DensityMatrixGateTest.MergeMultiply
  [       OK ] DensityMatrixGateTest.MergeMultiply (0 ms)
  [ RUN      ] DensityMatrixGateTest.MergeTensorProductAndMultiply
  [       OK ] DensityMatrixGateTest.MergeTensorProductAndMultiply (0 ms)
  [ RUN      ] DensityMatrixGateTest.RandomPauliMerge
  [       OK ] DensityMatrixGateTest.RandomPauliMerge (0 ms)
  [ RUN      ] DensityMatrixGateTest.RandomPauliRotationMerge
  [       OK ] DensityMatrixGateTest.RandomPauliRotationMerge (2 ms)
  [ RUN      ] DensityMatrixGateTest.RandomUnitaryMerge
  [       OK ] DensityMatrixGateTest.RandomUnitaryMerge (1 ms)
  [ RUN      ] DensityMatrixGateTest.RandomUnitaryMergeLarge
  [       OK ] DensityMatrixGateTest.RandomUnitaryMergeLarge (1 ms)
  [ RUN      ] DensityMatrixGateTest.RandomControlMergeSmall
  [       OK ] DensityMatrixGateTest.RandomControlMergeSmall (1 ms)
  [ RUN      ] DensityMatrixGateTest.RandomControlMergeLarge
  [       OK ] DensityMatrixGateTest.RandomControlMergeLarge (3 ms)
  [ RUN      ] DensityMatrixGateTest.MultiTarget
  [       OK ] DensityMatrixGateTest.MultiTarget (84 ms)
  [ RUN      ] DensityMatrixGateTest.MultiControlSingleTarget
  [       OK ] DensityMatrixGateTest.MultiControlSingleTarget (27 ms)
  [ RUN      ] DensityMatrixGateTest.SingleControlMultiTarget
  [       OK ] DensityMatrixGateTest.SingleControlMultiTarget (51 ms)
  [ RUN      ] DensityMatrixGateTest.MultiControlMultiTarget
  [       OK ] DensityMatrixGateTest.MultiControlMultiTarget (28 ms)
  [----------] 17 tests from DensityMatrixGateTest (302 ms total)

  [----------] 4 tests from ObservableTest
  [ RUN      ] ObservableTest.CheckExpectationValue
  [       OK ] ObservableTest.CheckExpectationValue (2 ms)
  [ RUN      ] ObservableTest.CheckParsedObservableFromOpenFermionFile
  [       OK ] ObservableTest.CheckParsedObservableFromOpenFermionFile (0 ms)
  [ RUN      ] ObservableTest.CheckParsedObservableFromOpenFermionText
  [       OK ] ObservableTest.CheckParsedObservableFromOpenFermionText (1 ms)
  [ RUN      ] ObservableTest.CheckSplitObservable
  [       OK ] ObservableTest.CheckSplitObservable (0 ms)
  [----------] 4 tests from ObservableTest (3 ms total)

  [----------] 2 tests from DensityMatrixObservableTest
  [ RUN      ] DensityMatrixObservableTest.CheckExpectationValue
  [       OK ] DensityMatrixObservableTest.CheckExpectationValue (1 ms)
  [ RUN      ] DensityMatrixObservableTest.CheckParsedObservableFromOpenFermionText
  [       OK ] DensityMatrixObservableTest.CheckParsedObservableFromOpenFermionText (1 ms)
  [----------] 2 tests from DensityMatrixObservableTest (2 ms total)

  [----------] 10 tests from DensityMatrixGeneralGateTest
  [ RUN      ] DensityMatrixGeneralGateTest.ProbabilisticGate
  [       OK ] DensityMatrixGeneralGateTest.ProbabilisticGate (0 ms)
  [ RUN      ] DensityMatrixGeneralGateTest.CPTPGate
  [       OK ] DensityMatrixGeneralGateTest.CPTPGate (0 ms)
  [ RUN      ] DensityMatrixGeneralGateTest.InstrumentGate
  [       OK ] DensityMatrixGeneralGateTest.InstrumentGate (0 ms)
  [ RUN      ] DensityMatrixGeneralGateTest.AdaptiveGate
  [       OK ] DensityMatrixGeneralGateTest.AdaptiveGate (0 ms)
  [ RUN      ] DensityMatrixGeneralGateTest.CheckProbabilisticGate
  [       OK ] DensityMatrixGeneralGateTest.CheckProbabilisticGate (26 ms)
  [ RUN      ] DensityMatrixGeneralGateTest.CheckCPTPMap
  [       OK ] DensityMatrixGeneralGateTest.CheckCPTPMap (1 ms)
  [ RUN      ] DensityMatrixGeneralGateTest.AmplitudeDampingTest
  [       OK ] DensityMatrixGeneralGateTest.AmplitudeDampingTest (0 ms)
  [ RUN      ] DensityMatrixGeneralGateTest.DepolarizingTest
  [       OK ] DensityMatrixGeneralGateTest.DepolarizingTest (0 ms)
  [ RUN      ] DensityMatrixGeneralGateTest.TwoQubitDepolarizingTest
  [       OK ] DensityMatrixGeneralGateTest.TwoQubitDepolarizingTest (0 ms)
  [ RUN      ] DensityMatrixGeneralGateTest.NoiseSampling
  [       OK ] DensityMatrixGeneralGateTest.NoiseSampling (0 ms)
  [----------] 10 tests from DensityMatrixGeneralGateTest (28 ms total)

  [----------] 3 tests from NoiseSimulatorTest
  [ RUN      ] NoiseSimulatorTest.Random_with_State_Test
  [       OK ] NoiseSimulatorTest.Random_with_State_Test (9 ms)
  [ RUN      ] NoiseSimulatorTest.Random_without_State_Test
  [       OK ] NoiseSimulatorTest.Random_without_State_Test (10 ms)
  [ RUN      ] NoiseSimulatorTest.H_gate_twice_test
  [       OK ] NoiseSimulatorTest.H_gate_twice_test (8 ms)
  [----------] 3 tests from NoiseSimulatorTest (27 ms total)

  [----------] 9 tests from StateTest
  [ RUN      ] StateTest.GenerateAndRelease
  [       OK ] StateTest.GenerateAndRelease (1 ms)
  [ RUN      ] StateTest.Sampling
  [       OK ] StateTest.Sampling (0 ms)
  [ RUN      ] StateTest.SetState
  [       OK ] StateTest.SetState (0 ms)
  [ RUN      ] StateTest.GetMarginalProbability
  [       OK ] StateTest.GetMarginalProbability (0 ms)
  [ RUN      ] StateTest.AddState
  [       OK ] StateTest.AddState (0 ms)
  [ RUN      ] StateTest.MultiplyCoef
  [       OK ] StateTest.MultiplyCoef (0 ms)
  [ RUN      ] StateTest.TensorProduct
  [       OK ] StateTest.TensorProduct (0 ms)
  [ RUN      ] StateTest.DropQubit
  [       OK ] StateTest.DropQubit (0 ms)
  [ RUN      ] StateTest.PermutateQubit
  [       OK ] StateTest.PermutateQubit (0 ms)
  [----------] 9 tests from StateTest (2 ms total)

  [----------] 6 tests from DensityMatrixTest
  [ RUN      ] DensityMatrixTest.GenerateAndRelease
  [       OK ] DensityMatrixTest.GenerateAndRelease (0 ms)
  [ RUN      ] DensityMatrixTest.Sampling
  [       OK ] DensityMatrixTest.Sampling (1 ms)
  [ RUN      ] DensityMatrixTest.SetState
  [       OK ] DensityMatrixTest.SetState (0 ms)
  [ RUN      ] DensityMatrixTest.GetMarginalProbability
  [       OK ] DensityMatrixTest.GetMarginalProbability (0 ms)
  [ RUN      ] DensityMatrixTest.AddState
  [       OK ] DensityMatrixTest.AddState (0 ms)
  [ RUN      ] DensityMatrixTest.MultiplyCoef
  [       OK ] DensityMatrixTest.MultiplyCoef (29 ms)
  [----------] 6 tests from DensityMatrixTest (30 ms total)

  [----------] Global test environment tear-down
  [==========] 87 tests from 9 test cases ran. (526 ms total)
  [  PASSED  ] 87 tests.
  Running main() from C:\Users\sg495\Documents\git\qulacs\qulacs\visualstudio\_deps\googletest_fetch-src\googletest\src\gtest_main.cc
  [==========] Running 85 tests from 7 test cases.
  [----------] Global test environment set-up.
  [----------] 7 tests from CircuitTest
  [ RUN      ] CircuitTest.CircuitBasic
  [       OK ] CircuitTest.CircuitBasic (170 ms)
  [ RUN      ] CircuitTest.CircuitOptimize
  [       OK ] CircuitTest.CircuitOptimize (18 ms)
  [ RUN      ] CircuitTest.RandomCircuitOptimize
  [       OK ] CircuitTest.RandomCircuitOptimize (18 ms)
  [ RUN      ] CircuitTest.RandomCircuitOptimizeLarge
  [       OK ] CircuitTest.RandomCircuitOptimizeLarge (18 ms)
  [ RUN      ] CircuitTest.SuzukiTrotterExpansion
  [       OK ] CircuitTest.SuzukiTrotterExpansion (37 ms)
  [ RUN      ] CircuitTest.RotateDiagonalObservable
  [       OK ] CircuitTest.RotateDiagonalObservable (4 ms)
  [ RUN      ] CircuitTest.SpecialGatesToString
  [       OK ] CircuitTest.SpecialGatesToString (1 ms)
  [----------] 7 tests from CircuitTest (266 ms total)

  [----------] 4 tests from CompatTest
  [ RUN      ] CompatTest.ApplyRandomOrderUnitary
CUSTOMBUILD : error : C:/Users/sg495/Documents/git/qulacs/qulacs/src/gpusim/update_ops_multi.cu:277, code: 701, reason: too many resources requested for launch [C:\Users\sg495\Documents\git\qulacs\qulacs\visualst
udio\test.vcxproj]
```

I also executed step 5, just in case. It reports the same error I get from python, output below.

```
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  csim_static.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\lib\csim_static.lib
  gpusim_static.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\lib\gpusim_static.lib
  cppsim_static.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\lib\cppsim_static.lib
  vqcsim_static.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\lib\vqcsim_static.lib
  qulacs.vcxproj -> C:\Users\sg495\Documents\git\qulacs\qulacs\bin\qulacs.cp39-win_amd64.pyd
  Traceback (most recent call last):
    File "C:\Users\sg495\Documents\git\qulacs\qulacs\python\test\test_qulacs.py", line 3, in <module>
      import qulacs
  ImportError: DLL load failed while importing qulacs: The specified module could not be found.
C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(209,5): error MSB6006: "cmd.exe" exited with code 1. [C:\Users\sg495\Documents\git\qulacs\qulacs
\visualstudio\pythontest.vcxproj]
```

**Comment by sg495** [comment]
Any further suggestions on what could be wrong here?

**Comment by corryvrequan** [comment]
I'm very sorry for late reply...
Thanks for the report on my suggestion.

A situation is still mysterious for me... At least we have not encountered the same situation.
While the following may not help, let me suggest a possible reason.

1.
Please try the same thing with `./script/build_msvc_2017.bat`, i.e., without GPU options.
I understand you already checked `pip install qulacs` works, but this command installs pre-compiled binary from PyPI, while the installation with GPU compiles source codes in installation. Thus, check with `./script/build_msvc_2017.bat` will clarify whether the problem comes from GPU/CUDA or not. 
If the problem happens even without GPU, I guess essential libraries of MSVC are not reachable from python executable, which may happen when the directory is shadowed other privilege PATH setting set by python.

2.
When you already installed qulacs with the wrong compilation options, it may conflict with the newly installed one.
Note that "qulacs-gpu" is not a plugin of "qulacs". "qulacs-gpu" contains the whole function of "qulacs".
Please check no qulacs is installed in your python environment.

**Comment by toddytharmonkey** [comment]
Hello, I have run across the same error also on Windows 10. I have gone through the steps with the ./script/build_msvc_2017.bat file, with both the GPU/non GPU version, but I haven't spotted any obvious errors during the process. (I am assuming an error would halt the process or throw flashy messages? Which didn't seem to happen.) 

I was thinking reason 2 may be the reason not allowing me to use qulacs-gpu.  I downloaded the default qulacs first via "pip install qulacs" and now it seems I cannot install any other version, either manually or by pip. Would it be possible to know how to fix this? I have already tried installing qulacs-gpu on fresh python installation to no avail. (I get the same error.) 

Any help would be most appreciated!
